flow/core/kernel/base_energy.py

```diff
@@ -12,10 +11,10 @@ class BaseEnergyModel(metaclass=ABCMeta):
     """
 
     def __init__(self, kernel):
-        self.k.env.vehicle = vehicle
+        self.k = kernel
 
     @abstractmethod
-    def get_instantaneous_power(self):
+    def get_instantaneous_power(self, parameter, accel, speed, grade):
```

Comment:
it looks like `parameter` is for the old_soc, and i guess you are saving this for any other future parameters you might want to pass? if so, i think the pythonic way to do this is using args and kwargs (i'm not super familiar but you can read about it here: https://book.pythontips.com/en/latest/args_and_kwargs.html)

you should put this parameter last, and have it default to None if it's not passed.